src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,226 @@
+import classNames from 'classnames';
+import { FC, PropsWithChildren } from 'react';
+export type PaginationProps = PropsWithChildren<{
+  className?: string;
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+  // Optional props
+  showIcon?: boolean;
+  showIconOnly?: boolean;
+  showOnlyPreviousAndNext?: boolean;
+  tableDataPagination?: boolean;
+}>;
+
+export const Pagination: FC<PaginationProps> = ({
+  currentPage,
+  totalPages,
+  onPageChange,
+  showIcon,
+  showIconOnly,
+  showOnlyPreviousAndNext,
+  tableDataPagination,
+}) => {
+  // Logic for displaying page numbers
+  const pageNumbers = [];
+  const startPage = Math.max(1, currentPage - 3);
+  const endPage = Math.min(currentPage + 3, totalPages);
+  for (let i = startPage; i <= endPage; i++) {
+    pageNumbers.push(i);
+  }
+  if (tableDataPagination) {
+    return (
```

Comment:
I'm seeing too many returns in the middle of the large block of codes, it kind of makes me confuse. I think that this is something that can be improved. Maybe you could create she inner components to handle those alternative templates?

Reply:
done